tools/osbuilder/node-builder/azure-linux/Makefile

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2024 Microsoft Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+all: clean package uvm
+
+all-confpods: clean package-confpods uvm-confpods
```

Comment:
IMO `make clean` shouldn't be called automatically, that should be done by the user, iff they want to. And we should make sure the `package*` and `uvm*` targets are idempotent.

Something very important to me when developing is being able to run `make all-confpods && make deploy-confpods` fast to iterate quickly. Right now `make all-confpods` takes 10 mins to run on a 4-vCPU node, probably because we're cleaning each time and hampering Rust incremental builds.